include/eve/module/core/regular/impl/shuffle_l0.hpp

```diff
@@ -0,0 +1,65 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//==================================================================================================
+#pragma once
+
+#include <algorithm>
+#include <span>
+
+#include <initializer_list>
+
+namespace eve::detail::idx
+{
+
+constexpr bool
+matches(const std::span<const std::ptrdiff_t> idxs, std::initializer_list<std::ptrdiff_t> pattern)
+{
+  if( idxs.size() != pattern.size() ) return false;
+
+  auto pattern_f = pattern.begin();
+  for( unsigned i = 0; i != idxs.size(); ++i )
+  {
+    if( idxs[i] == we_ ) continue;
+    if( idxs[i] != pattern_f[i] ) return false;
+  }
+
+  return true;
+}
+
+constexpr bool
+is_identity(std::span<const std::ptrdiff_t> idxs)
+{
+  return idxs.size() == 1 && (idxs[0] == 0 || idxs[0] == we_);
```

Comment:
identity can be of any size, so I guess we only test that because identity get super simplified right ?

Reply:
Yeah. I originally wrote an array but then like: why